app/controllers/posts_controller.rb

```diff
@@ -1,11 +1,35 @@
 class PostsController < ApplicationController
   def index
     @user = User.find(params[:user_id])
-    @posts = Post.where(author: @user)
+    @post = Post.all
   end
 
   def show
-    puts params
     @post = Post.find(params[:id])
+    @comments = @post.comments
+  end
+
+  def new
+    @post = Post.new
+  end
+
+  def create
+    @post = current_user.posts.new(post_params[:id])
+    respond_to do |format|
+      format.html do
+        if @post.save
+          redirect_to user_post_path(current_user, @post), notice: 'Post was successfully created.'
+          # redirect_to '/users/#{@post_new.user_id}/posts', messaage: 'Post was successfully created.'
+        else
+          render :new, status: 'Error: Post was not created.'
+        end
+      end
+    end
```

Comment:
You have a bug in your project when I create a new post

<img width="1089" alt="image" src="https://user-images.githubusercontent.com/7243132/200123583-031d35cb-9520-42d2-91e3-f961f4e4cf03.png">


**Follow my others comments to correct it.**